### PR TITLE
Workflow security

### PR DESCRIPTION
**Pull request by MariaShaposhnikova**

### CODE REVIEW

File: content/en/docs/howto/security/configure-workflow-security.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+3. In the **Properties** dialog box, open the **Access rules** tab and click **New** to create a rule for the HR role:
+
+    {{< figure src="/attachments/howto/logic-business-rules/workflow-how-to-configure/entity-properties.png" alt="Entity Properties" >}} 
+
+4. In the **New Access rule** dialog box, do the following:
+
+    1. In the **Rule applies to the following modules** section, select the **HR** role.
+
+    2. In the **Create and delete rights** section, select **Allow creating new objects**. This allows HR to start the workflow.
```
Comment (PaulJongenMx) [inline content/en/docs/howto/security/configure-workflow-security.md]: > This allows HR to start the workflow.

 => This allows HR to create a new EmployeeOnboarding object when a workflow is started

File: content/en/docs/howto/security/configure-workflow-security.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+    {{< figure src="/attachments/howto/logic-business-rules/workflow-how-to-configure/entity-properties.png" alt="Entity Properties" >}} 
+
+4. In the **New Access rule** dialog box, do the following:
+
+    1. In the **Rule applies to the following modules** section, select the **HR** role.
+
+    2. In the **Create and delete rights** section, select **Allow creating new objects**. This allows HR to start the workflow. 
+
+    3. In the **Member read and write rights** section, click **Set all to Read**. As the HR needs only to start the workflow, but not to change any employee information, **Read** rights are sufficient for all attributes of the entity.
```
Comment (PaulJongenMx) [inline content/en/docs/howto/security/configure-workflow-security.md]: Wondering if this is correct and will work: beeing able to create an object (point 3) but have no rights to write to it (point 4).

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## 2 Configuring Entity Access {#entity-access}
+
+If a user does not have an access to an entity, this means they have no access to entity data. You can allow users to **Create**, **Read**, **Write** and/or **Delete** objects of the entity. If a user should be able to only view the page with user tasks assigned to them without making any changes, the **Read** access is sufficient. If a user needs to add, change, and/or delete data, you should allow **Create**, **Write**, and **Delete** objects for the corresponding user role. You can also configure an XPath for detailed access rules. 
+
+For more information on the entity access, see the [Entity Access vs. Page Access](/refguide/security/#entity-vs-page-access) section in *Security* and the [Entity Access](/refguide/module-security/#entity-access) section in *Module Security*.
+
+## 3 Configuring Page Access {#page-access}
+
+For each user task you set a dedicated page where users can view their **Task inbox** and interact with tasks. These pages are also added to the menu bar, which is optimized so that only pages that the user has access to are visible.
```
Comment (PaulJongenMx) [inline content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md]: > For each user task you set a dedicated page where users can view their **Task inbox** and interact with tasks.

=> For each user task you set a dedicated page where users can execute the task. These task pages are called from the task inbox. The task inbox should be added to the menu bar. Pls make sure to set proper access to the task inbox pages and all the task pages.

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+## 3 Configuring Page Access {#page-access}
+
+For each user task you set a dedicated page where users can view their **Task inbox** and interact with tasks. These pages are also added to the menu bar, which is optimized so that only pages that the user has access to are visible. 
+
+The combination of entity access and page access makes sure that only dedicated users can open a user task page and view its data. 
+
+For more information on the page access, see the [Page Access](/refguide/module-security/#page-access) section in *Module Security*.
+
+### 4 Configuring User Assignment {#user-assignment}
```
Comment (PaulJongenMx) [inline content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md:34]: This section should be different. 

The WF entity WorkflowUserTask is used in the inbox and task pages and has 2 Xpath constaints (that are similar). Let's have a look at the one for role User:
[State = 'InProgress' and (System.WorkflowUserTask_TargetUsers = '[%CurrentUser%]' or System.WorkflowUserTask_Assignee= '[%CurrentUser%]') and System.WorkflowUserTask_Workflow/System.Workflow[State != 'Incompatible' and State != 'Failed']].

This means that the currenlty logged in user can only see WorkflowUserTask data if (s)he is the assinged user or is a targeted user and the state of the task is "In Progress" and the statet is not "Failed" and not "Incompatible".

In the workflow User task properties, you specify what user is targeted for the task (i.e. will see the task in the inbox) by writing an XPath expression or microflow. You can set filters there and specify that only a certain user role can receive a task in their inbox, for example, only users from an IT department.

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+If the user does not see the user task, check the following:
+
+1. Make sure they have at least the **Read** access for the entity set for the **WorkflowContext** parameter. Open the domain model > entity properties > **Access rules** to check that.
+
+2. Make sure that the user role has the page access for the page configured for the user task. Open the page > page properties > **Visible for** property to check that.
+
+3. Make sure that the XPath specified for the **User assignment** property does not restrict the user role in viewing the task. Open workflow > user task properties > **Assign task using** property to check it.
+
+4. Conditions specified in the XPath constraint of the **System.WorkflowUserTask** entity are met (you can view them in the **System** module > domain model > **System.WorkflowUserTask** entity properties > **Access rules** tab > **XPath constraint** tab). The following conditions are specified:
```
Comment (PaulJongenMx) [inline content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md]: THis point 4 is not something developers can change/correct of forget to configure. It's a note/FYI so the developer knows there is an xpath on the entity.

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md (hunk 3 above)
Comment (MariaShaposhnikova) [reply content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md]: Added

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md (hunk 4 above)
Comment (MariaShaposhnikova) [reply content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md:34]: Rewritten

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md (hunk 5 above)
Comment (MariaShaposhnikova) [reply content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md]: In the text, we say "check that conditions specified are met". Basically, the text does not tell them to change anything but points what conditions can be contradicted.

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md (hunk 4 above)
Comment (MariaShaposhnikova) [reply content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md:34]: @PaulJongenMx , I have rewritten it a bit to make it clear why we mention this.

File: content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md (hunk 5 above)
Comment (MariaShaposhnikova) [reply content/en/docs/refguide/modeling/application-logic/workflows/workflow-security.md]: I have transferred information from this section to the previous one and made this point shorter.

File: content/en/docs/howto/security/configure-workflow-security.md (hunk 2 above)
Comment (MariaShaposhnikova) [reply content/en/docs/howto/security/configure-workflow-security.md]: @PaulJongenMx , I double-checked with Arjen, he says that if they don't need to fill information in, **Read** should be sufficient.

**Pull request merged**